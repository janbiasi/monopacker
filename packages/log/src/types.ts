export type LogLevel = 'debug' | 'info' | 'warn' | 'error' | 'success';

export type Logger = {
	[key in LogLevel]: (message: string) => void;
};
