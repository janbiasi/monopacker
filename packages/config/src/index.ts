export { Config, ResolverConfig, PackConfig, PackageManager } from './types';
export { getConfig } from './resolve';
export { validateConfig } from './validator';
export { defaultFilesToCopy, defaultIgnorePatterns } from './defaults';
