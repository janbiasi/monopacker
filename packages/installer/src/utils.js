// @ts-check

const { readdir } = require('fs');
const { resolve, join } = require('path');
const { exec } = require('child_process');

const ROOT = resolve(process.cwd());

/**
 * Get all packed dependencies from a monopacked package repository
 * @param {string} rootDir			Root path to the npm package
 * @returns {Promise<string[]>} 	List of files
 */
exports.getMonopackedDependencies = rootDir => {
	rootDir = resolve(rootDir);

	return new Promise((resolve, reject) => {
		readdir(rootDir, (err, files) => {
			if (err) {
				return reject(err);
			}
			resolve(files);
		});
	});
};

/**
 * Install a list of tarballs
 * @param {string[]} packedPaths	Path list of monopacker packed tarballs
 */
exports.installFromMonopackedTarballs = packedPaths => {
	return Promise.all(
		packedPaths.map(packedPath => {
			return new Promise((resolve, reject) => {
				exec(`npm install ${packedPath}`, (err, stdout, stderr) => {
					if (err) {
						return reject(stderr);
					}
					resolve(stdout);
				});
			});
		})
	);
};
