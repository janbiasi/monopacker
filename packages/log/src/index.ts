export * from './default';
export { LogLevel, LoggableLevel, loggableLevels } from './level';
export { createLogger, getLogStack } from './logger';
export { Logger } from './types';
export { Namespace, namespaces } from './namespace';
