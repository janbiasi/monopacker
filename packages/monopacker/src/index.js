"use strict";
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
Object.defineProperty(exports, "__esModule", { value: true });
// expose all types
__export(require("./types"));
// expose adapters and blueprint for custom ones
var adapter_1 = require("./adapter");
exports.Adapter = adapter_1.Adapter;
exports.AdapterLerna = adapter_1.AdapterLerna;
// expose main packer
var Packer_1 = require("./Packer");
exports.Packer = Packer_1.Packer;
// expose taper
var Taper_1 = require("./Taper");
exports.Taper = Taper_1.Taper;
// expose helpers
var debug_hooks_1 = require("./helper/debug-hooks");
exports.useDebugHooks = debug_hooks_1.useDebugHooks;
//# sourceMappingURL=index.js.map