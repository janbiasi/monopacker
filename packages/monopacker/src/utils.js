"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
var path_1 = require("path");
var crypto_1 = require("crypto");
var execa = require("execa");
exports.execa = execa;
var _rimraf = require("rimraf");
var fs = require("fs-extra");
exports.fs = fs;
var glob = require("glob");
var matcher = require("matcher");
exports.matcher = matcher;
var ncp_1 = require("ncp");
function findDuplicatesInArray(array) {
    var object = {};
    var result = [];
    array.forEach(function (item) {
        if (!object[item])
            object[item] = 0;
        object[item] += 1;
    });
    for (var prop in object) {
        if (object[prop] >= 2) {
            result.push(prop);
        }
    }
    return result;
}
exports.findDuplicatesInArray = findDuplicatesInArray;
function rimraf(pathName, options) {
    if (options === void 0) { options = {}; }
    return new Promise(function (resolve, reject) {
        _rimraf(pathName, options, function (err) {
            if (err) {
                return reject(err);
            }
            resolve();
        });
    });
}
exports.rimraf = rimraf;
function copyDir(source, destination, opts) {
    if (opts === void 0) { opts = {
        dereference: true
    }; }
    return new Promise(function (resolve, reject) {
        ncp_1.ncp(source, destination, opts, function (err) {
            if (err) {
                return reject(err);
            }
            resolve();
        });
    });
}
exports.copyDir = copyDir;
function getLernaPackages2(cwd) {
    return __awaiter(this, void 0, void 0, function () {
        var stdout;
        return __generator(this, function (_a) {
            switch (_a.label) {
                case 0: return [4 /*yield*/, execa('lerna', ['ls', '--all', '--json'], {
                        cwd: cwd
                    })];
                case 1:
                    stdout = (_a.sent()).stdout;
                    return [2 /*return*/, JSON.parse(stdout)];
            }
        });
    });
}
exports.getLernaPackages2 = getLernaPackages2;
function asyncForEach(array, callback) {
    return __awaiter(this, void 0, void 0, function () {
        var index;
        return __generator(this, function (_a) {
            switch (_a.label) {
                case 0:
                    if (!Array.isArray(array)) {
                        return [2 /*return*/, Promise.resolve()];
                    }
                    index = 0;
                    _a.label = 1;
                case 1:
                    if (!(index < array.length)) return [3 /*break*/, 4];
                    return [4 /*yield*/, callback(array[index], index)];
                case 2:
                    _a.sent();
                    _a.label = 3;
                case 3:
                    index++;
                    return [3 /*break*/, 1];
                case 4: return [2 /*return*/];
            }
        });
    });
}
exports.asyncForEach = asyncForEach;
function extractDependencies(dependencyLike, filter) {
    if (dependencyLike === void 0) { dependencyLike = {}; }
    if (filter === void 0) { filter = function () { return true; }; }
    return Object.keys(dependencyLike)
        .filter(filter)
        .reduce(function (prev, curr) {
        var _a;
        return (__assign({}, prev, (_a = {}, _a[curr] = dependencyLike[curr], _a)));
    }, {});
}
exports.extractDependencies = extractDependencies;
function readJSON(f) {
    return __awaiter(this, void 0, void 0, function () {
        var contents, err_1;
        return __generator(this, function (_a) {
            switch (_a.label) {
                case 0:
                    _a.trys.push([0, 2, , 3]);
                    return [4 /*yield*/, fs.readFile(f)];
                case 1:
                    contents = _a.sent();
                    return [2 /*return*/, JSON.parse(contents.toString())];
                case 2:
                    err_1 = _a.sent();
                    return [2 /*return*/, {}];
                case 3: return [2 /*return*/];
            }
        });
    });
}
exports.readJSON = readJSON;
function loadJson(f) {
    try {
        return JSON.parse(fs.readFileSync(f, 'utf8') || '');
    }
    catch (err) {
        return null;
    }
}
function getLernaPackages(root) {
    var patterns = (function () {
        try {
            return loadJson(root + "/lerna.json").packages;
        }
        catch (err) { }
        try {
            return loadJson(root + "/package.json").workspaces;
        }
        catch (err) { }
        return [];
    })() || [];
    var modules = patterns.reduce(function (acc, patt) {
        var found = glob.sync(patt, { cwd: root }).filter(function (f) { return !f.match(/node_modules/); });
        return acc.concat(found);
    }, []);
    var subModules = modules.reduce(function (acc, f) {
        try {
            var fname = root + "/" + f;
            var stat = fs.statSync(fname);
            fname = stat.isDirectory() ? fname : path_1.dirname(fname);
            var pkgJson = path_1.join(fname, "package.json");
            // const pkgStat = fs.statSync(pkgJson);
            var pkg_1 = loadJson(pkgJson);
            var ref_1 = path_1.resolve(path_1.relative(root, fname).replace(/[\\]+/g, '/'));
            var isDuplicate = acc.filter(function (m) { return m.location === ref_1; });
            if (!isDuplicate.length && pkg_1 && pkg_1 !== null) {
                if (!pkg_1.name || !pkg_1.version) {
                    throw new Error("Invalid package " + (pkg_1.name || '<unknown>') + " in " + ref_1 + ": name or version missing");
                }
                acc.push({
                    name: pkg_1.name,
                    version: pkg_1.version || '*',
                    description: pkg_1.description || '',
                    private: !!pkg_1.private,
                    location: ref_1
                });
            }
        }
        catch (err) {
            /* istanbul ignore next */
            throw err || 'Unable to list lerna modules';
        }
        return acc;
    }, []);
    return subModules;
}
exports.getLernaPackages = getLernaPackages;
function countMsg(countable, singular, plural) {
    plural = plural || singular + "s";
    var size = -1;
    if (Array.isArray(countable)) {
        size = countable.length;
    }
    else {
        size = Object.keys(countable).length;
    }
    return size === 1 ? "1 " + singular : size + " " + plural;
}
exports.countMsg = countMsg;
function displayPath(base, toDisplay) {
    return path_1.resolve(toDisplay).replace(path_1.resolve(base), '~');
}
exports.displayPath = displayPath;
function createHash(value) {
    return crypto_1.createHash('sha256')
        .update(value)
        .digest('hex');
}
exports.createHash = createHash;
function createIntegrityHash(version, analytics) {
    return createHash(version +
        JSON.stringify(__assign({}, analytics, { dependencies: __assign({}, analytics.dependencies, { internal: analytics.dependencies.internal.map(function (entry) { return entry.name + "@" + entry.version + "|" + !!entry.private; }) }) })));
}
exports.createIntegrityHash = createIntegrityHash;
exports.pkg = require('../package.json');
//# sourceMappingURL=utils.js.map