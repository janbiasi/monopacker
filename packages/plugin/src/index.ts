export const test = 10;
