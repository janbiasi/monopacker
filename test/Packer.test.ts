import { resolve, join } from 'path';
import { Packer } from '../src/Packer';
import { fs } from '../src/utils';
import { HookPhase } from '../src/types';
import { createTestPackerFor, createResolverWithin } from './utils';
import { readFile, existsSync } from 'fs-extra';

const BASIC_CWD = resolve(__dirname, 'fixtures/basic');

const createTestPackerForBasic = (hooks?: Packer['hooks']) => createTestPackerFor(BASIC_CWD, 'packages/main', hooks);

const resolveInTarget = createResolverWithin(join(BASIC_CWD, 'temp'));

describe('Packer', () => {
	describe('Lerna', () => {
		describe('core functions', () => {
			it('should generate an artificial package correctly', async () => {
				const packer = createTestPackerForBasic();
				await packer.pack();
				const generatedPkg = require(resolveInTarget('package.json'));
				expect(generatedPkg).toBeTruthy();
				expect(generatedPkg).toMatchSnapshot();
				expect(generatedPkg.name).toEqual('@fixture/main-packed');
				expect(generatedPkg.dependencies).toStrictEqual({
					'ansi-styles': '4.0.0', // from subsub
					debug: '*', // from main
					smallest: '1.0.1', // from sub
					'supports-color': '7.0.0' // from subsubsub
				});
			});

			it('should copy the source files correctly', async () => {
				const packer = createTestPackerForBasic();
				await packer.pack();
				const contents = require(resolveInTarget('src', 'test.js'));
				expect(contents).toEqual('Hello world');
			});

			it('should copy sub-modules correctly', async () => {
				const fakeHook = jest.fn(async () => Promise.resolve());
				const packer = createTestPackerForBasic({
					[HookPhase.INIT]: [fakeHook],
					[HookPhase.PRECOPY]: [fakeHook],
					[HookPhase.POSTCOPY]: [fakeHook],
					[HookPhase.PREANALYZE]: [fakeHook],
					[HookPhase.POSTANALYZE]: [fakeHook],
					[HookPhase.PRELINK]: [fakeHook],
					[HookPhase.POSTLINK]: [fakeHook],
					[HookPhase.PACKED]: [fakeHook]
				});
				// TODO: call time was 10 before, we removed install as default
				await packer.pack();
				expect(fakeHook).toBeCalledTimes(7);
			});

			it.skip('should install all external modules', async () => {
				const packer = createTestPackerForBasic();
				await packer.pack();

				const debugExists = await fs.pathExists(resolveInTarget('node_modules', 'debug'));
				expect(debugExists).toBe(true);

				const msExists = await fs.pathExists(resolveInTarget('node_modules', 'ms'));
				expect(msExists).toBe(true);

				const smallestExists = await fs.pathExists(resolveInTarget('node_modules', 'smallest'));
				expect(smallestExists).toBe(true);

				const arrayExists = await fs.pathExists(resolveInTarget('node_modules', 'to-array'));
				expect(arrayExists).toBe(true);
			});

			it('should generate a valid registry file', async () => {
				const packer = createTestPackerForBasic();
				await packer.pack();

				let contents = await readFile(resolveInTarget('.monopacker', 'monopacker.registry.json'));
				const stringContents = contents.toString();
				expect(stringContents).toMatchSnapshot();

				const list = JSON.parse(stringContents);
				expect(list.length).toEqual(3);
				expect(list).toStrictEqual([
					'./.monopacker/fixture-sub-0.1.3.tgz',
					'./.monopacker/fixture-subsub-0.1.3.tgz',
					'./.monopacker/fixture-subsubsub-0.1.3.tgz'
				]);
			});

			it('should pack all internal modules to tarballs', async () => {
				const packer = createTestPackerForBasic();
				await packer.pack();

				const fixtureSubExists = existsSync(resolveInTarget('.monopacker', 'fixture-sub-0.1.3.tgz'));
				const fixtureSubSubExists = existsSync(resolveInTarget('.monopacker', 'fixture-subsub-0.1.3.tgz'));
				const fixtureSubSubSubExists = existsSync(
					resolveInTarget('.monopacker', 'fixture-subsubsub-0.1.3.tgz')
				);

				expect(fixtureSubExists).toBe(true);
				expect(fixtureSubSubExists).toBe(true);
				expect(fixtureSubSubSubExists).toBe(true);
			});

			it('should generate correct metadata for the packed bundle', async () => {
				const packer = createTestPackerForBasic();
				await packer.pack();
				const generatedPkg = require(resolveInTarget('package.json'));
				expect(generatedPkg).toBeDefined();
				expect(generatedPkg.monopacker).toBeDefined();
				expect(generatedPkg.monopacker).toMatchSnapshot();
			});

			it('should generate a hash for the packed bundle', async () => {
				const packer = createTestPackerForBasic();
				await packer.pack();
				const generatedPkg = require(resolveInTarget('package.json'));
				expect(generatedPkg).toBeDefined();
				expect(generatedPkg.monopacker).toBeDefined();
				expect(generatedPkg.monopacker.hash).toBeDefined();
				expect(generatedPkg.monopacker.hash.length).toBeGreaterThan(0);
				expect(generatedPkg.monopacker.hash).toMatchSnapshot();
			});
		});
	});
});
