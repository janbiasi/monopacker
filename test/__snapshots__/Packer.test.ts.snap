// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Packer Lerna multitree dependencies should analyze multitree dependencies correctly 1`] = `
Object {
  "dependencies": Object {
    "external": Object {},
    "internal": Array [
      Object {
        "description": "",
        "name": "@fixture/multitree-ab",
        "private": false,
        "version": "1.0.0",
      },
      Object {
        "description": "",
        "name": "@fixture/multitree-a",
        "private": false,
        "version": "1.0.0",
      },
      Object {
        "description": "",
        "name": "@fixture/multitree-b",
        "private": false,
        "version": "1.0.0",
      },
    ],
    "peer": Object {
      "ms": "2.1.2",
      "smallest": "1.0.1",
    },
  },
  "graph": Object {
    "@fixture/multitree-a": Object {
      "smallest": "1.0.1",
    },
    "@fixture/multitree-ab": Object {},
    "@fixture/multitree-b": Object {
      "ms": "2.1.2",
    },
    "@fixture/multitree-c": Object {
      "external": Object {
        "ms": "2.1.2",
        "smallest": "1.0.1",
      },
      "internal": Object {
        "@fixture/multitree-a": "1.0.0",
        "@fixture/multitree-ab": "1.0.0",
        "@fixture/multitree-b": "1.0.0",
      },
    },
  },
  "integrity": "a1e8d10d61d9c191c853d7e797a7887f2ea8601414903b00587461e9f1e1a0e2",
}
`;

exports[`Packer Lerna multitree dependencies should pack multitree apps correctly 1`] = `
Object {
  "dependencies": Object {
    "ms": "2.1.2",
    "smallest": "1.0.1",
  },
  "description": "",
  "monopacker": Object {
    "hash": "a1e8d10d61d9c191c853d7e797a7887f2ea8601414903b00587461e9f1e1a0e2",
    "linked": Object {
      "@fixture/multitree-a": "1.0.0",
      "@fixture/multitree-ab": "1.0.0",
      "@fixture/multitree-b": "1.0.0",
    },
    "version": "1",
  },
  "name": "@fixture/multitree-c-packed",
  "scripts": Object {
    "postinstall": "npm i --no-scripts ./.monopacker/fixture-multitree-ab-1.0.0.tgz ./.monopacker/fixture-multitree-a-1.0.0.tgz ./.monopacker/fixture-multitree-b-1.0.0.tgz",
  },
  "version": "1.0.0",
}
`;

exports[`Packer Lerna should aggregate any analytics 1`] = `
Object {
  "dependencies": Object {
    "external": Object {
      "debug": "*",
    },
    "internal": Array [
      Object {
        "description": "",
        "name": "@fixture/sub",
        "private": false,
        "version": "0.1.3",
      },
      Object {
        "description": "",
        "name": "@fixture/subsub",
        "private": false,
        "version": "0.1.3",
      },
      Object {
        "description": "",
        "name": "@fixture/subsubsub",
        "private": false,
        "version": "0.1.3",
      },
    ],
    "peer": Object {
      "ansi-styles": "4.0.0",
      "debug": "*",
      "smallest": "1.0.1",
      "supports-color": "7.0.0",
    },
  },
  "graph": Object {
    "@fixture/main": Object {
      "external": Object {
        "ansi-styles": "4.0.0",
        "debug": "*",
        "smallest": "1.0.1",
        "supports-color": "7.0.0",
      },
      "internal": Object {
        "@fixture/sub": "0.1.3",
        "@fixture/subsub": "0.1.3",
        "@fixture/subsubsub": "0.1.3",
      },
    },
    "@fixture/sub": Object {
      "smallest": "1.0.1",
    },
    "@fixture/subsub": Object {
      "debug": "*",
    },
    "@fixture/subsubsub": Object {
      "ansi-styles": "4.0.0",
      "supports-color": "7.0.0",
    },
  },
  "integrity": "c47f8432f2003d16eaa5afe09720adf81ef7b7b45388e687e7f480c20829cf99",
}
`;

exports[`Packer Lerna should generate a hash for the packed bundle 1`] = `"c47f8432f2003d16eaa5afe09720adf81ef7b7b45388e687e7f480c20829cf99"`;

exports[`Packer Lerna should generate a monopacker.analytics.json file 1`] = `
Object {
  "dependencies": Object {
    "external": Object {
      "debug": "*",
    },
    "internal": Array [
      Object {
        "description": "",
        "name": "@fixture/sub",
        "private": false,
        "version": "0.1.3",
      },
      Object {
        "description": "",
        "name": "@fixture/subsub",
        "private": false,
        "version": "0.1.3",
      },
      Object {
        "description": "",
        "name": "@fixture/subsubsub",
        "private": false,
        "version": "0.1.3",
      },
    ],
    "peer": Object {
      "ansi-styles": "4.0.0",
      "debug": "*",
      "smallest": "1.0.1",
      "supports-color": "7.0.0",
    },
  },
  "graph": Object {
    "@fixture/main": Object {
      "external": Object {
        "ansi-styles": "4.0.0",
        "debug": "*",
        "smallest": "1.0.1",
        "supports-color": "7.0.0",
      },
      "internal": Object {
        "@fixture/sub": "0.1.3",
        "@fixture/subsub": "0.1.3",
        "@fixture/subsubsub": "0.1.3",
      },
    },
    "@fixture/sub": Object {
      "smallest": "1.0.1",
    },
    "@fixture/subsub": Object {
      "debug": "*",
    },
    "@fixture/subsubsub": Object {
      "ansi-styles": "4.0.0",
      "supports-color": "7.0.0",
    },
  },
  "integrity": "c47f8432f2003d16eaa5afe09720adf81ef7b7b45388e687e7f480c20829cf99",
}
`;

exports[`Packer Lerna should generate a monopacker.analytics.json file 2`] = `
Object {
  "dependencies": Object {
    "external": Object {
      "debug": "*",
    },
    "internal": Array [
      Object {
        "description": "",
        "name": "@fixture/sub",
        "private": false,
        "version": "0.1.3",
      },
      Object {
        "description": "",
        "name": "@fixture/subsub",
        "private": false,
        "version": "0.1.3",
      },
      Object {
        "description": "",
        "name": "@fixture/subsubsub",
        "private": false,
        "version": "0.1.3",
      },
    ],
    "peer": Object {
      "ansi-styles": "4.0.0",
      "debug": "*",
      "smallest": "1.0.1",
      "supports-color": "7.0.0",
    },
  },
  "graph": Object {
    "@fixture/main": Object {
      "external": Object {
        "ansi-styles": "4.0.0",
        "debug": "*",
        "smallest": "1.0.1",
        "supports-color": "7.0.0",
      },
      "internal": Object {
        "@fixture/sub": "0.1.3",
        "@fixture/subsub": "0.1.3",
        "@fixture/subsubsub": "0.1.3",
      },
    },
    "@fixture/sub": Object {
      "smallest": "1.0.1",
    },
    "@fixture/subsub": Object {
      "debug": "*",
    },
    "@fixture/subsubsub": Object {
      "ansi-styles": "4.0.0",
      "supports-color": "7.0.0",
    },
  },
  "integrity": "c47f8432f2003d16eaa5afe09720adf81ef7b7b45388e687e7f480c20829cf99",
}
`;

exports[`Packer Lerna should generate an artificial package correctly 1`] = `
Object {
  "dependencies": Object {
    "ansi-styles": "4.0.0",
    "debug": "*",
    "smallest": "1.0.1",
    "supports-color": "7.0.0",
  },
  "description": "",
  "monopacker": Object {
    "hash": "c47f8432f2003d16eaa5afe09720adf81ef7b7b45388e687e7f480c20829cf99",
    "linked": Object {
      "@fixture/sub": "0.1.3",
      "@fixture/subsub": "0.1.3",
      "@fixture/subsubsub": "0.1.3",
    },
    "version": "1",
  },
  "name": "@fixture/main-packed",
  "scripts": Object {
    "postinstall": "npm i --no-scripts ./.monopacker/fixture-sub-0.1.3.tgz ./.monopacker/fixture-subsub-0.1.3.tgz ./.monopacker/fixture-subsubsub-0.1.3.tgz",
  },
  "version": "0.2.7",
}
`;

exports[`Packer Lerna should generate correct metadata for the packed bundle 1`] = `
Object {
  "hash": "c47f8432f2003d16eaa5afe09720adf81ef7b7b45388e687e7f480c20829cf99",
  "linked": Object {
    "@fixture/sub": "0.1.3",
    "@fixture/subsub": "0.1.3",
    "@fixture/subsubsub": "0.1.3",
  },
  "version": "1",
}
`;
